lib/vtls/openssl.c

```diff
@@ -3081,8 +3137,22 @@ static CURLcode ossl_connect_step1(struct Curl_easy *data,
   }
 #endif
 
+  if(ca_info_pem) {
+    result = load_cacert_from_memory(backend->ctx, ca_info_pem);
+    if(result) {
+      if(verifypeer) {
+        /* Fail if we insist on successfully verifying the server. */
+        failf(data, "error setting certificate from memory");
+        return result;
+      }
+      /* Only warning if no certificate verification is required. */
+      infof(data, "error setting certificate file, continuing anyway\n");
```

Comment:
1. It didn't "set" a file though, did it? It failed to import a CA cert from the given PEM buffer, right?
2. Also, is that really a reason for failing the connection? What if a CA bundle on file is also used?
3. If the function failed due to out of memory I think it should fail no matter what